src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -389,9 +389,14 @@ private void UndoActionGroup(XmlElement actionGroup)
                         ModelBase toBeUpdated = undoClient.GetModelForElement(element);
                         if (toBeUpdated != null)
                         {
-
                             RecordActionInternal(newGroup, toBeUpdated, modelActionType);
                             undoClient.ReloadModel(element);
+
+                            if(toBeUpdated is Graph.Notes.NoteModel)
+                            {
+                                var noteModel = (Graph.Notes.NoteModel)toBeUpdated;
+                                noteModel.TryToSubscribeUndoNote();
```

Comment:
Is it because the `pinnedNode` is not a serializable property so we have to make an exception here?

Reply:
Again, I would love to see if we can remove the special case here